skrf/network.py

```diff
@@ -2059,7 +2059,8 @@ def copy(self) -> Network:
         ntwk = Network(z0=self.z0, s_def=self.s_def, comments=self.comments)
 
         ntwk._s = self.s.copy()
-        ntwk.frequency = self.frequency
+        ntwk.frequency._f = self.frequency._f.copy()
+        ntwk.frequency.unit = self.frequency.unit
```

Comment:
Good catch, otherwise it would be a shallow copy with refernence to the old network.
In there future we will hopefully manage to get `frequency` immutable, so shallow copies would be possible without any risk.

Reply:
The assignment `ntwk.frequency = self.frequency` is not a shallow copy, as noted in https://github.com/scikit-rf/scikit-rf/pull/1096#discussion_r1645306232. The `@frequency.setter` decorator automatically performs a deep copy.